client/service/src/task_manager.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+//! Substrate service tasks management module.
+
+use std::{
+	result::Result, sync::Arc,
+	task::{Poll, Context},
+	borrow::Cow, pin::Pin,
+};
+use exit_future::Signal;
+use log::{debug, error};
+use futures::{
+	Future, FutureExt, Stream,
+	future::select, channel::mpsc,
+	compat::*,
+	task::{Spawn, FutureObj, SpawnError},
+};
+
+/// Alias for a an implementation of `futures::future::Executor`.
+pub type TaskExecutor = Arc<dyn Spawn + Send + Sync>;
+
+/// Type alias for service task executor (usually runtime).
+pub type ServiceTaskExecutor = Arc<dyn Fn(Pin<Box<dyn Future<Output = ()> + Send>>) + Send + Sync>;
+
+/// Type alias for the task scheduler.
+pub type TaskScheduler = mpsc::UnboundedSender<(Pin<Box<dyn Future<Output = ()> + Send>>, Cow<'static, str>)>;
+
+/// An handle for spawning tasks in the service.
+#[derive(Clone)]
+pub struct SpawnTaskHandle {
+	sender: TaskScheduler,
+	on_exit: exit_future::Exit,
+}
+
+impl SpawnTaskHandle {
+	/// Spawns the given task with the given name.
+	pub fn spawn(&self, name: impl Into<Cow<'static, str>>, task: impl Future<Output = ()> + Send + 'static) {
```

Comment:
You could possibly have this take an `impl Future<Output = T>` instead if you wanted to avoid using `.map(drop)` when calling it.

Reply:
Hmm I am not sure I want something to be dropped silently, and this `T` cannot be returned in any way.